implementations/rust/ockam/ockam_command/src/space/delete.rs

```diff
@@ -36,37 +37,97 @@ pub struct DeleteCommand {
 
 impl DeleteCommand {
     pub fn run(self, options: CommandGlobalOpts) {
-        node_rpc(rpc, (options, self));
+        node_rpc(run_impl, (options, self));
     }
 }
 
-async fn rpc(ctx: Context, (opts, cmd): (CommandGlobalOpts, DeleteCommand)) -> miette::Result<()> {
-    run_impl(&ctx, opts, cmd).await
+async fn run_impl(
+    ctx: Context,
+    (opts, cmd): (CommandGlobalOpts, DeleteCommand),
+) -> miette::Result<()> {
+    DeleteTui::run(ctx, opts, cmd).await
 }
 
-async fn run_impl(
-    ctx: &Context,
+pub struct DeleteTui {
+    ctx: Context,
     opts: CommandGlobalOpts,
-    cmd: DeleteCommand,
-) -> miette::Result<()> {
-    if opts
-        .terminal
-        .confirmed_with_flag_or_prompt(cmd.yes, "Are you sure you want to delete this space?")?
-    {
-        let space_id = opts.state.spaces.get(&cmd.name)?.config().id.clone();
-        let node = InMemoryNode::start(ctx, &opts.state).await?;
+    space_name: Option<String>,
+    controller: Controller,
+}
+
+impl DeleteTui {
+    pub async fn run(
+        ctx: Context,
+        opts: CommandGlobalOpts,
+        cmd: DeleteCommand,
+    ) -> miette::Result<()> {
+        let node = InMemoryNode::start(&ctx, &opts.state).await?;
         let controller = node.create_controller().await?;
-        controller.delete_space(ctx, space_id).await?;
+        let tui = Self {
+            ctx,
+            opts,
+            space_name: cmd.name,
+            controller,
+        };
+        tui.delete().await
+    }
+}
+
+#[ockam_core::async_trait]
+impl DeleteCommandTui for DeleteTui {
+    const ITEM_NAME: &'static str = "space";
+
+    fn cmd_arg_item_name(&self) -> Option<&str> {
+        self.space_name.as_deref()
+    }
+
+    fn cmd_arg_delete_all(&self) -> bool {
+        false
+    }
+
+    fn cmd_arg_confirm_deletion(&self) -> bool {
+        false
```

Comment:
I think we should add this feature to this command. It would be as simple as adding the `yes` argument as it's done in [other commands](https://github.com/build-trust/ockam/blob/54bb211c0f1eb20af324cc5e37b38f3a43ff9817/implementations/rust/ockam/ockam_command/src/relay/delete.rs#L31-L33).

Reply:
Makes sense!